shoop/core/cache.py

```diff
@@ -0,0 +1,161 @@
+# -*- coding: utf-8 -*-
+"""
+Utilities for caching, with versioned caching and automatic timeout determination.
+
+Versioning works by way of "namespaces". Namespaces are the first colon-separated part
+of cache keys.
+
+For instance, the cache keys `price:10`, `price:20`, and `price` all belong to the
+`price` namespace and can be invalidated with one `bump_version("price")` call.
+
+The versions themselves are stored within the cache, within the `_version` namespace.
+(As an implementation detail, this allows one to invalidate _all_ versioned keys by
+bumping the version of `_version`. Very meta!)
+"""
+import time
+from django.conf import settings
+from django.utils.encoding import force_text
+from django.core.cache import caches
+from django.core.signals import request_finished
+from threading import local
+
+DEFAULT_CACHE_DURATIONS = {}
```

Comment:
Hmm.. maybe you're right. It's better to have the defaults somewhere else than in the settings and so the default setting would be just empty dict as you proposed.

Can you add a comment to `SHOOP_CACHE_DURATIONS` or `SHOOP_DEFAULT_CACHE_DURATION` setting that there is also `shoop.core.cache.DEFAULT_CACHE_DURATIONS` so that people configuring the cache durations know why some cache duration does not have `SHOOP_DEFAULT_CACHE_DURATION` as its value even though it's not in the `SHOOP_CACHE_DURATIONS` dict?